includes/wikia/parser/templatetypes/TemplateTypesParser.class.php

```diff
@@ -35,6 +36,104 @@ public static function onFetchTemplateAndTitle( &$text, &$finalTitle ) {
 	}
 
 	/**
+	 * @desc change template wikitext according to template type
+	 *
+	 * @param $templateTitle
+	 * @param $templateWikitext
+	 * @return bool
+	 */
+	public static function onBraceSubstitution( $templateTitle, &$templateWikitext ) {
+		wfProfileIn( __METHOD__ );
+		$title = Title::newFromText( $templateTitle, NS_TEMPLATE );
+
+		if ( self::templateReadyToProcess( $templateWikitext ) && self::isValidTitle( $title ) ) {
+			$type = self::getTemplateType( $title );
+			if ( $type == AutomaticTemplateTypes::TEMPLATE_CONTEXT_LINK ) {
+				$templateWikitext = self::handleContextLinksTemplate( $templateWikitext );
+			}
+		}
+
+		wfProfileOut( __METHOD__ );
+
+		return true;
+	}
+
+	/**
+	 * @desc return template type for a given template title object
+	 *
+	 * @param $title
+	 * @return string
+	 */
+	private static function getTemplateType( $title ) {
+		global $wgCityId;
+		$type = ( new ExternalTemplateTypesProvider( new \TemplateClassificationService ) )
+			->getTemplateTypeFromTitle( $wgCityId, $title );
+
+		return $type;
+	}
+
+	private static function templateReadyToProcess( $templateWikitext ) {
+		global $wgEnableTemplateTypesParsing, $wgArticleAsJson;
+
+		return $wgEnableTemplateTypesParsing
+			&& $wgArticleAsJson
+			&& !empty( $templateWikitext );
+	}
+
+	/**
+	 * @desc sanitize context-link template content
+	 *
+	 * @param $wikitext
+	 * @return string
+	 */
+	private static function handleContextLinksTemplate( $wikitext ) {
+		$wikitext = self::sanitizeContextLinkWikitext( $wikitext );
+		$wikitext = self::wrapContextLink( $wikitext );
+
+		return $wikitext;
+	}
+
+	/**
+	 * @desc remove all non-link and non-text elements of context-link wikitext
+	 *
+	 * @param $wikitext string context-link template wikitext
+	 * @return string
+	 */
+	public static function sanitizeContextLinkWikitext( $wikitext ) {
+		//remove any custom HTML tags
+		$wikitext = strip_tags( $wikitext );
+		//remove list and indent elements from the beginning of line
+		$wikitext = preg_replace( '/^[:#* \n]+/', '', $wikitext );
+		//remove all bold and italics from all of template content
+		$wikitext = preg_replace( '/\'{2,}/', '', $wikitext );
+		//remove all headings from all of template content
+		$wikitext = str_replace( '=', '', $wikitext );
+		//remove all newlines from the middle of the template text.
```

Comment:
I know it does not matter in HTML, but what do you think about folding multiple spaces?